src/status_im/contexts/wallet/wallet_connect/events/sessions.cljs

```diff
@@ -26,116 +26,59 @@
-      :db (assoc db :wallet-connect/sessions sessions')})))
+   (log/info "WalletConnect sessions loaded successfully")
+   {:db (assoc db :wallet-connect/sessions sessions)}))
 
 (rf/reg-event-fx
- :wallet-connect/fetch-persisted-sessions-fail
+ :wallet-connect/get-sessions-error
  (fn [_ [error]]
-   (log/info "Wallet Connect fetch persisted sessions failed" error)
-   {:fx [[:dispatch [:wallet-connect/fetch-active-sessions]]]}))
-
-(rf/reg-event-fx
- :wallet-connect/fetch-persisted-sessions
- (fn [{:keys [now]} _]
-   (let [current-timestamp (quot now 1000)]
-     {:fx [[:json-rpc/call
-            [{:method     "wallet_getWalletConnectActiveSessions"
-              ;; NOTE: This is the activeSince timestamp to avoid expired sessions
-              :params     [current-timestamp]
-              :on-success [:wallet-connect/fetch-persisted-sessions-success]
-              :on-error   [:wallet-connect/fetch-persisted-sessions-fail]}]]]})))
+   (log/error "WalletConnect sessions failed to load" error)
+   {:fx [[:dispatch
+          [:toasts/upsert
+           {:type :negative
+            :text (i18n/label :t/wallet-connect-connections-error)}]]]}))
 
 (rf/reg-event-fx
- :wallet-connect/persist-session
- (fn [_ [session-info]]
-   (let [redirect-url (-> session-info
-                          (js->clj :keywordize-keys true)
-                          (data-store/get-dapp-redirect-url))]
-     {:fx [[:json-rpc/call
-            [{:method     "wallet_addWalletConnectSession"
-              :params     [(js/JSON.stringify session-info)]
-              :on-success (fn []
-                            (log/info "Wallet Connect session persisted")
-                            (rf/dispatch [:wallet-connect/fetch-persisted-sessions])
-                            (rf/dispatch [:wallet-connect/redirect-to-dapp redirect-url]))
-              :on-error   #(log/info "Wallet Connect session persistence failed" %)}]]]})))
+ :wallet-connect/on-new-session
+ (fn [{:keys [db]} [new-session]]
+   {:db (update db
+                :wallet-connect/sessions
+                (fn [sessions]
+                  (->> new-session
+                       sessions/sdk-session->db-session
+                       (conj sessions))))}))
 
 (rf/reg-event-fx
- :wallet-connect/disconnect-persisted-session
+ :wallet-connect/delete-session
  (fn [{:keys [db]} [topic]]
-   (log/info "Removing session from persistance and state" topic)
    {:db (update db
                 :wallet-connect/sessions
                 (fn [sessions]
                   (->> sessions
                        (remove #(= (:topic %) topic))
-                       (into []))))
-    :fx [[:json-rpc/call
-          [{:method     "wallet_disconnectWalletConnectSession"
-            :params     [topic]
-            :on-success #(log/info "Wallet Connect session disconnected")
-            :on-error   #(log/info "Wallet Connect session persistence failed" %)}]]]}))
+                       (into []))))}))
```

Comment:
@seanstrom so here is the cause of the bug. This event was called when disconnecting a dapp and when syncing persisted dapps. We shouldn't be removing active sessions from the db when syncing the persisted dapps. The faulty "sync" was likely caused by the expiry date check, but hard to tell the cause since it's difficult to reproduce.